File: examples/undocumented/python_static/tools/load.py
#!/usr/bin/env python

from numpy import double, fromfile

class LoadMatrix:
	def __init__(self):
		pass


	def load_numbers(self, filename):
		matrix=fromfile(filename, sep=' ')
		# whole matrix is 1-dim now, so reshape
		matrix=matrix.reshape(2, len(matrix)/2)
		#matrix=matrix.reshape(len(matrix)/2, 2)

		return matrix


	def load_dna(self, filename):
		fh=open(filename, 'r');
		matrix=[]

		# handle row/column brain damage
		for line in fh:
			matrix.append(line[:-1])
		fh.close()

		return matrix


	def load_cubes(self, filename):
		fh=open(filename, 'r');
		matrix=[]

		for line in fh:
			matrix.append(line.split(' ')[0][:-1])

		fh.close()

		return matrix


	def load_labels(self, filename):
		return fromfile(filename, dtype=double, sep=' ')

File: examples/undocumented/lua_modular/load.lua
function load_numbers(fname)
	f = io.open(fname)
	if (f == nil) then
		print("Cannot open ", fname)
	end
	matrix = {}
	while true do
		local	 num = f:read("*number")
		if not num then break end
		table.insert(matrix, num)
	end
	f:close()
	n = #matrix
	result = {}
	for i = 1, 2 do
		result[i] = {}
		for j = 1, n/2 do
			result[i][j] = matrix[(i-1) * (n/2) + j]
		end
	end
	return result
end

function load_labels(fname)
	f = io.open(fname)
	if (f == nil) then
		print("Cannot open ", fname)
	end
	matrix = {}
	while true do
		local	 num = f:read("*number")
		if not num then break end
		table.insert(matrix, num)
	end
	f:close()
	return matrix
end

function load_dna(fname)
	f = io.open(fname)
	if (f == nil) then
		print("Cannot open file:", fname)
	end
	matrix = {}
	while true do
		print("love")
		local	 num = f:read("*line")
		if not num then break end
		table.insert(matrix, num)
	end
	f:close()
	for k, v in pairs(matrix) do print (v) end
	return matrix
end

function load_cubes(fname)
	f = io.open(fname)
	if (f == nil) then
		print("Cannot open file:", fname)
	end
	matrix = {}
	while true do
		local	 num = f:read("*line")
		if not num then break end
		table.insert(matrix, num)
	end
	f:close()
	for k, v in pairs(matrix) do print (v) end
	return matrix
end
